phpBB/phpbb/avatar/driver/upload.php

```diff
@@ -161,6 +161,16 @@ public function process_form($request, $template, $user, $row, &$error)
 			return false;
 		}
 
+		$ext = substr(strrchr($row['avatar'], '.'), 1);
+		if ($ext && $ext !== $file->get('extension'))
+		{
+			$filename = $this->phpbb_root_path . $this->config['avatar_path'] . '/' . $this->config['avatar_salt'] . '_' . $row['id'] . '.' . $ext;
```

Comment:
Ok, I have been looking at this more, and what I would like to do is to call
$this->delete($row, $error);
And have $error be passed by reference and take a default value of array().
By doing this, in case an error would happen in delete method, the result could be propagated back to the user, and the success return would only be sent in case no error happened on delete either.  I would move this block to above, L156.
If you agree, I will update this fix, and the other ticket as well to be in line (prototype for delete function).